consensus/reactor.go

```diff
@@ -696,30 +696,32 @@ func (conR *ConsensusReactor) gossipVotesForHeight(logger log.Logger, rs *cstype
 			return true
 		}
 	}
-	// If there are prevotes to send...
-	if prs.Step <= cstypes.RoundStepPrevote && prs.Round != -1 && prs.Round <= rs.Round {
+
+	if prs.Step <= cstypes.RoundStepCommit && prs.Round != -1 && prs.Round <= rs.Round {
```

Comment:
dont think `Step < RoundStepCommit` is  meaningful here, since it should be always true.

I guess the idea is before, we were only sending prevotes if you still needed them.

But now we're saying you always need them, because you need to be able to see a polka if there was one.

Is that right ? 

I'd also think there's no reason not to just send the prevotes and the precommits if there are any to send. @jaekwon  ?

Reply:
That's right. We need to be able to see a polka until the end of the round, so we are interested in Prevotes all the time. Only in case you are in RoundStepCommit you don't need Prevotes as you decided. So we can remove `Step < RoundStepCommit` top level, and maybe add it to Prevote related condition. And as you said, I would not constrain us in sending one kind of messages in one pass; I would send the prevotes and precommits if there are some to be sent.